### PR TITLE
[CI] Add some extra params to configure the test templates.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: build-tools/automation/yaml-templates/stage-msbuild-emulator-tests.yaml [hunk 1]
```diff
@@ -1,14 +1,17 @@
 # Runs MSBuild tests against a device running on macOS
 
 parameters:
+  stageName: msbuilddevice_tests
   job_name: 'mac_dotnetdevice_tests'
+  depensOn: mac_build
```
Comment (pjcollins) [inline build-tools/automation/yaml-templates/stage-msbuild-emulator-tests.yaml:6]: Small typo here is breaking the default `dependsOn` case, the same issue is also present below.

**Review by pjcollins — CHANGES_REQUESTED** [review]

**Review by pjcollins — COMMENTED** [review]
I am wondering if it makes sense to put test jobs for MAUI dependencies in the unified pipeline... I don't think we plan on moving test jobs out of our pipelines, and we're already struggling with machine resources for many of our test jobs. I think it may be better for the unified pipeline to run all of the MAUI tests and any MAUI integration tests we have, rather than duplicate existing test runs from its dependencies.

Maybe we can continue to run non-MAUI tests in the individual product repos, and instead add some manual intervention / sign off to the end stages of the unified pipeline where product owners could approve specific commits?

**Comment by dalexsoto** [comment]
My take on this and we all can discuss further, is that if we are resource constrained we definitely need to address this by requesting the hardware we need in the labs. 

I think there is a ton of value having these tests in the unified pipeline as we will have one stop for all the tests done in the bits that we intend to ship and insert so there is no doubt on the quality of our releases. Resource wise this should not be much of a burden as we can initially use the unified pipeline for our release branches. 

I agree that we can have test rings and we can split testing as we see fit, but the testing capability must be present in the mega pipeline since the beginning else we will keep piling backlog and this work is supposed to help to increase the confidence on our builds all the way.

File: build-tools/automation/yaml-templates/stage-msbuild-emulator-tests.yaml (hunk 1 above)
Comment (pjcollins) [reply build-tools/automation/yaml-templates/stage-msbuild-emulator-tests.yaml:6]: We will also need to consider the build caching aspect of the unified pipeline, these jobs will need to be skipped if the Android commit has already been built.

**Comment by pjcollins** [comment]
> I think there is a ton of value having these tests in the unified pipeline as we will have one stop for all the tests done in the bits that we intend to ship and insert so there is no doubt on the quality of our releases. Resource wise this should not be much of a burden as we can initially use the unified pipeline for our release branches.

I don't think duplicating existing PR checks is going to have an impact on our confidence in the quality of a given set of builds, we have a plenty of gaps to fill elsewhere in this regard. The machine pool availability concern is primarily a result of the fact that the unified pipeline runs for every commit to every release branch of every product repo. While that run is batched, it will still increase demand on our infrastructure quite a bit (there were ~16 builds today alone). I'm having a bit of difficulty seeing how the value outweighs the cost, even if we had infinite capacity.

File: build-tools/automation/yaml-templates/stage-msbuild-emulator-tests.yaml (hunk 1 above)
Comment (mandel-macaque) [reply build-tools/automation/yaml-templates/stage-msbuild-emulator-tests.yaml:6]:
```suggestion
  dependsOn: mac_build
```
Got it, I could not see it

File: build-tools/automation/yaml-templates/stage-msbuild-tests.yaml [hunk 2]
```diff
@@ -1,12 +1,15 @@
 # Runs MSBuild tests that do not require devices on Mac/Windows
 
 parameters:
+  stageName: msbuild_dotnet
   stageCondition: succeeded()
+  depensOn: mac_build
```
Comment (mandel-macaque) [inline build-tools/automation/yaml-templates/stage-msbuild-tests.yaml:6]:
```suggestion
  dependsOn: mac_build
```
Same

Comment (mandel-macaque) [inline build-tools/automation/yaml-templates/stage-msbuild-tests.yaml:6]:
```suggestion
  dependsOn: mac_build
```
Same

**Comment by mandel-macaque** [comment]
> > I think there is a ton of value having these tests in the unified pipeline as we will have one stop for all the tests done in the bits that we intend to ship and insert so there is no doubt on the quality of our releases. Resource wise this should not be much of a burden as we can initially use the unified pipeline for our release branches.
> 
> I don't think duplicating existing PR checks is going to have an impact on our confidence in the quality of a given set of builds, we have a plenty of gaps to fill elsewhere in this regard. The machine pool availability concern is primarily a result of the fact that the unified pipeline runs for every commit to every release branch of every product repo. While that run is batched, it will still increase demand on our infrastructure quite a bit (there were ~16 builds today alone). I'm having a bit of difficulty seeing how the value outweighs the cost, even if we had infinite capacity.

We can have the possibility and ignore them, that is not an issue. We just want to have the template added to the pipeline, if the resources are an issue, it can be disabled the same way it is in the macios tests.

**Pull request merged**